Main.py

```diff
@@ -157,33 +218,54 @@ def Button_2fa_Clicked():
         AccountArea.setLayout(Layout)
         AccountArea.exec()
 
+
 @QtCore.Slot()
 def pair():
-    subprocess.run('idevicepair pair',shell=True)
+    subprocess.run("idevicepair pair", shell=True)
+
 
 @QtCore.Slot()
 def restart_daemon():
-    subprocess.run(f'killall {AltServer}',shell=True)
-    subprocess.run('idevicepair pair',shell=True)
-    subprocess.run(f'{AltServer} &> /dev/null &',shell=True)
+    subprocess.run(f"killall {AltServer}", shell=True)
+    subprocess.run("idevicepair pair", shell=True)
+    subprocess.run(f"{AltServer} &> /dev/null &", shell=True)
+
 
 @QtCore.Slot()
 def check_update():
+    # Only check update for Ubuntu/Debian
+    if not shutil.which("dpkg"):
+        Not_deb_distro_msg_box = QMessageBox()
+        Not_deb_distro_msg_box.setText(
+            "Update is only supported for Ubuntu/Debian distro"
```

Comment:
Why restrict others distro which not debian/ubuntu based,they still can do dpkg,for arch based they just need sudo pacman -S dpkg

Reply:
They technically can, but really really really shouldn't. And I don't want people to ruin their OS because of this. Different package manager will have different dependency tracking